src/ledger/LedgerManagerImpl.h

```diff
@@ -42,58 +42,64 @@ class BasicWork;
     };
 
-    Application& mApp;
-    std::unique_ptr<XDROutputFileStream> mMetaStream;
-    std::unique_ptr<XDROutputFileStream> mMetaDebugStream;
-    std::weak_ptr<BasicWork> mFlushAndRotateMetaDebugWork;
-    std::filesystem::path mMetaDebugPath;
+    // Any state that apply needs to access through the app connector should go
+    // here, at very least just to make it clear what is being accessed by which
+    // threads. We may try to further encapsulate it.
+    struct ApplyState
+    {
+        // Latest Soroban config during apply (should not be used outside of
+        // application, as it may be in half-valid state). Note that access to
+        // this variable is not synchronized, since it should only be used by
+        // one thread (main or ledger close).
+        std::shared_ptr<SorobanNetworkConfig> mSorobanNetworkConfig;
+    };
 
-  private:
-    // Cache LCL state, accessible only from main thread
-    LedgerHeaderHistoryEntry mLastClosedLedger;
-
-    // Read-only Soroban network configuration, accessible by main thread only.
-    // This config has to match the read-only ledger state snapshot managed by
-    // LedgerManager. Today this is only accessed by the main thread, but it can
-    // be used by multiple threads as long snapshot consistency
-    // requirement is preserved (though synchronization should be added in this
-    // case)
-    std::shared_ptr<SorobanNetworkConfig const> mSorobanNetworkConfigReadOnly;
-
-    // Latest Soroban config during apply (should not be used outside of
-    // application, as it may be in half-valid state). Note that access to this
-    // variable is not synchronized, since it should only be used by one thread
-    // (main or ledger close).
-    std::shared_ptr<SorobanNetworkConfig> mSorobanNetworkConfigForApply;
-
-    // Cache most recent HAS, accessible only from main thread
-    HistoryArchiveState mLastClosedLedgerHAS;
+    struct LedgerApplyMetrics
+    {
+        medida::Timer& mTransactionApply;
+        medida::Histogram& mTransactionCount;
+        medida::Histogram& mOperationCount;
+        medida::Histogram& mPrefetchHitRate;
+        medida::Timer& mLedgerClose;
+        medida::Buckets& mLedgerAgeClosed;
+        medida::Counter& mLedgerAge;
+        medida::Counter& mTransactionApplySucceeded;
+        medida::Counter& mTransactionApplyFailed;
+        medida::Counter& mSorobanTransactionApplySucceeded;
+        medida::Counter& mSorobanTransactionApplyFailed;
+        medida::Meter& mMetaStreamBytes;
+        medida::Timer& mMetaStreamWriteTime;
+        LedgerApplyMetrics(medida::MetricsRegistry& registry);
+    };
+
+    // This state is private to the apply thread and holds work-in-progress
+    // that gets accessed via the AppConnector, from inside transactions.
+    ApplyState mApplyState;
 
+    // Cached LCL state output from last apply (or loaded from DB on startup).
+    LedgerState mLastClosedLedgerState;
```

Comment:
If we wanted to get _really_ pedantic, we could do

```
class LedgerStateGetter {
    LedgerState state;

public:
    LedgerState& get() {
        assert(main);
        return state;
    }

} mLastClosedLedgerState;
```

but I don't feel too strongly.

Reply:
eh .. it gets constructed on the apply thread and passed back to main. I think I am happy with the assert on main only accessing the LCL state.